crypto/tpe/src/lib.rs

```diff
@@ -130,14 +131,25 @@ fn hash_to_message(
     ephemeral_pk: &G1Affine,
     commitment: &sha256::Hash,
 ) -> G2Affine {
-    let message = (
-        "FEDIMINT_TPE_BLS12_381_MESSAGE",
-        *encrypted_point,
-        *ephemeral_pk,
-        *commitment,
-    );
-
-    let seed = message.consensus_hash::<sha256::Hash>().to_byte_array();
+    let mut engine = sha256::HashEngine::default();
+
+    engine
+        .write_all("FEDIMINT_TPE_BLS12_381_MESSAGE".as_bytes())
+        .expect("Writing to a hash engine cannot fail");
+
+    engine
+        .write_all(encrypted_point)
+        .expect("Writing to a hash engine cannot fail");
+
+    engine
+        .write_all(&ephemeral_pk.to_compressed())
+        .expect("Writing to a hash engine cannot fail");
+
+    engine
+        .write_all(commitment.as_byte_array())
+        .expect("Writing to a hash engine cannot fail");
+
+    let seed = sha256::Hash::from_engine(engine).to_byte_array();
```

Comment:
Any chance we can get rid of the `fedimint-core` dep this way?

Reply:
No because we need it to derive the consensus encoding.